src/components/ColumnSettingsList.ts

```diff
@@ -182,7 +182,7 @@ export const ColumnSettingsList = ({ items, onChange, toggleVisibility }: Column
   );
 
   const renderItem = (item, index) => {
-    return h(SortableItem, { id: item.id, name: item.name, index: getPosition(item.id) }, [
+    return h(SortableItem, { id: item.name, name: item.name, index: getPosition(item.id) }, [
```

Comment:
I think `ColumnSettingsList`, and therefore this use of `SortableItem`, is only used in the Column Settings Modal that this PR addresses, so I don't think this change should affect anything else.  The items don't have ids, only names, and having an undefined id was causing some problems.

Reply:
The items do have IDs like '0', '1', '2', since they're part of an array. I reintroduced the IDs, and everything worked fine, except for a complication with `isDragging`. When loading saved settings, the ColumnSettingsList items became semi-transparent due to changes in opacity.

I agree with your assessment that this component is only used in this instance. However, if any issues arise in the future, we may need to refactor it.

Reply:
Right - that was the problem I was trying to fix.  The issue is that the ids seem to be based on their position in the list of *checked* boxes - so they would change when you checked or unchecked them, which I think was the issue.